manim/__main__.py

```diff
@@ -1,14 +1,168 @@
-from . import extract_scene
-from . import config
+import inspect
+import itertools as it
+import os
+import platform
+import subprocess as sp
+import sys
+import traceback
+import importlib.util
+
+from .config import file_writer_config
+from .scene.scene import Scene
+from .utils.sounds import play_error_sound
+from .utils.sounds import play_finish_sound
 from . import constants
+from .logger import logger
+
+
+def open_file_if_needed(file_writer):
+    if file_writer_config["quiet"]:
+        curr_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    open_file = any([
+        file_writer_config["preview"],
+        file_writer_config["show_file_in_finder"]
+    ])
+    if open_file:
+        current_os = platform.system()
+        file_paths = []
+
+        if file_writer_config["save_last_frame"]:
+            file_paths.append(file_writer.get_image_file_path())
+        if file_writer_config["write_to_movie"]:
+            file_paths.append(file_writer.get_movie_file_path())
+
+        for file_path in file_paths:
+            if current_os == "Windows":
+                os.startfile(file_path)
+            else:
+                commands = []
+                if current_os == "Linux":
+                    commands.append("xdg-open")
+                elif current_os.startswith("CYGWIN"):
+                    commands.append("cygstart")
+                else:  # Assume macOS
+                    commands.append("open")
+
+                if file_writer_config["show_file_in_finder"]:
+                    commands.append("-R")
+
+                commands.append(file_path)
+
+                # commands.append("-g")
+                FNULL = open(os.devnull, 'w')
+                sp.call(commands, stdout=FNULL, stderr=sp.STDOUT)
+                FNULL.close()
+
+    if file_writer_config["quiet"]:
```

Comment:
Brief optimization: replace repeated `file_writer_config["quiet"]` getitem ops by just one op
```suggestion
    is_quiet = file_writer_config["quiet"]
    if is_quiet:
        curr_stdout = sys.stdout
        sys.stdout = open(os.devnull, "w")

    open_file = any([
        file_writer_config["preview"],
        file_writer_config["show_file_in_finder"]
    ])
    if open_file:
        current_os = platform.system()
        file_paths = []

        if file_writer_config["save_last_frame"]:
            file_paths.append(file_writer.get_image_file_path())
        if file_writer_config["write_to_movie"]:
            file_paths.append(file_writer.get_movie_file_path())

        for file_path in file_paths:
            if current_os == "Windows":
                os.startfile(file_path)
            else:
                commands = []
                if current_os == "Linux":
                    commands.append("xdg-open")
                elif current_os.startswith("CYGWIN"):
                    commands.append("cygstart")
                else:  # Assume macOS
                    commands.append("open")

                if file_writer_config["show_file_in_finder"]:
                    commands.append("-R")

                commands.append(file_path)

                # commands.append("-g")
                FNULL = open(os.devnull, 'w')
                sp.call(commands, stdout=FNULL, stderr=sp.STDOUT)
                FNULL.close()

    if is_quiet:
```

Reply:
Do you think this is an _measurable_ optimization at all? If we are able to measure a tangible optimization given by this change, then I'm happy to add it. If not, then I'm not so sure. After all, the current version is I think a little bit clearer. In your proposed change there is one more variable that the reader/developer must remember where it came from.

Reply:
well... in this sense, it's actually better to keep it as a variable in order to absolutely refute the possibility of the value changing, and also `is_quiet` is pretty clear on its own (as clear as the other alternative would be at least). But the optimization is pretty much irrelevant; the other ones i proposed are more significant

Reply:
I think it is unnecessary to define this variable just for the purpose of guaranteeing immutability. In that case, we should have `config.py` export `config` as a frozendict.